anosikedoris493/Sports-Betting-Platform
# Decentralized Autonomous Sports Betting Platform

## Overview

This pull request introduces a Decentralized Autonomous Sports Betting Platform implemented as a smart contract using Clarity. The platform allows users to create betting events, place bets, report results, and claim winnings in a trustless and transparent manner.

## Key Features

1. Event Creation: Users can create betting events with multiple options.
2. Bet Placement: Users can place bets on specific options within an event.
3. Result Reporting: An designated oracle can report the result of an event.
4. Winnings Claim: Users can claim their winnings after an event is concluded.
5. Odds Calculation: The contract calculates odds based on the total betting pool and individual option pools.

## Changes Made

- Created `sports-betting.clar` contract with the following main functions:
  - `create-event`
  - `place-bet`
  - `report-result`
  - `claim-winnings`
- Implemented helper functions for odds calculation and winning amount determination.
- Added data maps for storing event and user bet information.
- Implemented access control for owner-only and oracle-only functions.

## Testing Instructions

1. Ensure you have Clarinet installed on your system.
2. Navigate to the project directory in your terminal.
3. Run the following command to check the contract for errors:
